src/components/look-controls.js

```diff
@@ -182,22 +185,36 @@ module.exports.Component = registerComponent('look-controls', {
    * Mouse-drag only enabled if HMD is not active.
    */
   updateOrientation: function () {
-    var el = this.el;
     var hmdEuler = this.hmdEuler;
+    var object3D = this.el.object3D;
     var pitchObject = this.pitchObject;
     var yawObject = this.yawObject;
     var sceneEl = this.el.sceneEl;
+    var poseMatrix = this.poseMatrix;
+    var posePosition = this.posePosition;
+    var poseRotation = this.poseRotation;
+    var poseScale = this.poseScale;
+
+    // WebXR API updates applies headset pose to the object3D matrixWorld internally.
+    // Reflect values back on position, rotation, scale so setAttribute returns expected values.
+    if (sceneEl.is('vr-mode') && utils.device.isWebXRAvailable) {
+      poseMatrix.elements = sceneEl.renderer.vr.getCameraPose().poseModelMatrix;
+      poseMatrix.decompose(posePosition, poseRotation, poseScale);
+      object3D.position.copy(posePosition);
```

Comment:
posePosition, poseRotation variables not needed. Just decompose right into the object3D position and rotation. poseScale can be defined in the IIFE scope.